plugins/outputs/opensearch/sample.conf

```
@@ -0,0 +1,79 @@
+# Configuration for Opensearch to send metrics to.
+[[outputs.opensearch]]
+  ## The full HTTP endpoint URL for your Opensearch instance
+  ## Multiple urls can be specified as part of the same cluster,
+  ## this means that only ONE of the urls will be written to each interval
+  urls = [ "http://node1.os.example.com:9200" ] # required.
+  ## Opensearch client timeout, defaults to "5s" if not set.
+  timeout = "5s"
+  ## Set to true to ask Opensearch a list of all cluster nodes,
+  ## thus it is not necessary to list all nodes in the urls config option
+  enable_sniffer = false
+  ## Set to true to enable gzip compression
+  enable_gzip = false
+  ## Set the interval to check if the Opensearch nodes are available
+  ## Setting to "0s" will disable the health check (not recommended in production)
+  health_check_interval = "10s"
+  ## Set the timeout for periodic health checks.
+  # health_check_timeout = "1s"
+  ## HTTP basic authentication details.
+  ## HTTP basic authentication details
+  # username = "telegraf"
+  # password = "mypassword"
+  ## HTTP bearer token authentication details
+  # auth_bearer_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"
+
+  ## Index Config
+  ## The target index for metrics (Opensearch will create if it not exists).
+  ## You can use the date specifiers below to create indexes per time frame.
+  ## The metric timestamp will be used to decide the destination index name
+  # %Y - year (2016)
+  # %y - last two digits of year (00..99)
+  # %m - month (01..12)
+  # %d - day of month (e.g., 01)
+  # %H - hour (00..23)
+  # %V - week of the year (ISO week) (01..53)
+  ## Additionally, you can specify a tag name using the notation {{tag_name}}
+  ## which will be used as part of the index name. If the tag does not exist,
+  ## the default tag value will be used.
+  # index_name = "telegraf-{{host}}-%Y.%m.%d"
+  # default_tag_value = "none"
+  index_name = "telegraf-%Y.%m.%d" # required.
```

If we are going to show "telegraf-%Y.%m.%d" as a default, then it either needs to be the default or needs to show "" in the config and maybe suggest this value. I am inclined to say we just set that as the default if it is sane, but if it is not I would rather us just use an empty string. What do you think?

```suggestion
  ## Index Name
  ## The target index for metrics (Opensearch will create if it not exists).
  ## You can use the date specifiers below to create indexes per time frame.
  ## The metric timestamp will be used to decide the destination index name
  ##   %Y - year (2016)
  ##   %y - last two digits of year (00..99)
  ##   %m - month (01..12)
  ##   %d - day of month (e.g., 01)
  ##   %H - hour (00..23)
  ##   %V - week of the year (ISO week) (01..53)
  index_name = "telegraf-%Y.%m.%d"
  
  ## Default Index Tag Value
  ## Additionally, you can specify a tag name using the notation {{tag_name}}
  ## which will be used as part of the index name. If the tag does not exist,
  ## the default tag value will be used.
  # index_name = "telegraf-{{host}}-%Y.%m.%d"
  # default_tag_value = "none"
```

---

done